types/wlr_idle_inhibit_v1.c

```diff
@@ -142,13 +156,16 @@ void wlr_idle_inhibit_v1_destroy(struct wlr_idle_inhibit_manager_v1 *idle_inhibi
 	struct wlr_idle_inhibitor_v1 *inhibitor;
 	struct wlr_idle_inhibitor_v1 *tmp;
 	wl_list_for_each_safe(inhibitor, tmp, &idle_inhibit->inhibitors, link) {
-		wl_resource_destroy(inhibitor->resource);
+		struct wl_resource *resource = inhibitor->resource;
```

Comment:
I think it's fine to destroy resources when tearing down the whole manager. Gobals like this aren't supposed to be plug-and-play anyway, and making resources inert introduces complexity for all interfaces of the protocol and is error-prone.